src/app/auth/auth.service.ts

```diff
@@ -192,4 +224,14 @@ export class AuthService {
   updateProfile(data: UserProfile): Observable<any> {
     return this.request.post(api.setProfile, data);
   }
-}
\ No newline at end of file
+
+  saveRegistration(data: registerData): Observable<any> {
+    data.user_id = this.storage.get("hash").id;
+    data.key = this.storage.get("hash").key;
+    return this.request.post(api.register, data);
+  }
+
+  verifyRegistration(data: verifyParams): Observable<any> {
+    return this.request.post(api.verifyRegistration, data);
```

Comment:
This API call is not triggered, get error `ERROR TypeError: Cannot read property 'headers' of undefined` instead

Reply:
The error i'm getting `Unexpected token u in JSON at position 0`. this error also getting in the login. I try to fixed it. but for that i need to edit requestService. so i think it's better chaw edit that. 
If you found any reason in my code please let me know.